nano/node/socket.cpp

```diff
@@ -324,28 +342,43 @@ size_t network_prefix)
 bool nano::server_socket::limit_reached_for_incoming_subnetwork_connections (std::shared_ptr<nano::socket> const & new_connection)
 {
 	debug_assert (strand.running_in_this_thread ());
-	if (node.flags.disable_max_peers_per_subnetwork || nano::transport::is_ipv4_or_v4_mapped_address (new_connection->remote.address ()))
+
+	if (!new_connection->remote_endpoint ().has_value ())
+	{
+		return false;
+	}
+
+	if (node.flags.disable_max_peers_per_subnetwork || nano::transport::is_ipv4_or_v4_mapped_address (new_connection->remote_endpoint ()->address ()))
 	{
 		// If the limit is disabled, then it is unreachable.
 		// If the address is IPv4 we don't check for a network limit, since its address space isn't big as IPv6 /64.
 		return false;
 	}
+
 	auto const counted_connections = socket_functions::count_subnetwork_connections (
 	connections_per_address,
-	new_connection->remote.address ().to_v6 (),
+	new_connection->remote_endpoint ()->address ().to_v6 (),
 	node.network_params.network.ipv6_subnetwork_prefix_for_limiting);
+
 	return counted_connections >= node.network_params.network.max_peers_per_subnetwork;
 }
 
 bool nano::server_socket::limit_reached_for_incoming_ip_connections (std::shared_ptr<nano::socket> const & new_connection)
 {
 	debug_assert (strand.running_in_this_thread ());
+
 	if (node.flags.disable_max_peers_per_ip)
 	{
 		// If the limit is disabled, then it is unreachable.
 		return false;
 	}
-	auto const address_connections_range = connections_per_address.equal_range (new_connection->remote.address ());
+
+	if (!new_connection->remote_endpoint ().has_value ())
```

Comment:
If we're going to check has_value here and assume that below, this should be a copy of the endpoint.